build.gradle

```diff
@@ -32,10 +32,14 @@ version                    "${file("VERSION").text.trim()}" // read from require
 description                = "Elastic Integration filter"
 // ===========================================================================
 
+def JAVA_VERSION = 17
+
 java {
-    sourceCompatibility = 17
-    targetCompatibility = 17
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(JAVA_VERSION))
+    }
```

Comment:
review note: addition to spike diff, we declared java plugin (twice?!) but never applied so that Java target/source compatibility options weren't applied as well. If we have JDK 21 default on our local, plugin build against ES main treeish works which is not our expectation.